subsys/net/l2/ieee802154/ieee802154_fragment.h

```diff
@@ -18,19 +18,51 @@
 
 #include <net/net_pkt.h>
 
+#include "ieee802154_frame.h"
+
+#ifdef CONFIG_NET_L2_IEEE802154_FRAGMENT
+
+struct ieee802154_fragment_ctx {
+	struct net_buf *frag;
+	u8_t *pos;
+	u16_t pkt_size;
+	u16_t processed;
+	u8_t hdr_diff;
+	u8_t offset;
+};
+
+static inline bool ieee802154_fragment_is_needed(struct net_pkt *pkt,
+						 u8_t ll_hdr_size)
+{
+	return (net_pkt_get_len(pkt) + ll_hdr_size > IEEE802154_MTU);
+}
+
+static inline
+void ieee802154_fragment_ctx_init(struct ieee802154_fragment_ctx *ctx,
+				  struct net_pkt *pkt, u16_t hdr_diff)
+{
+	ctx->frag = pkt->frags;
+	ctx->pos = ctx->frag->data;
+	ctx->hdr_diff = hdr_diff;
+	ctx->pkt_size = net_pkt_get_len(pkt) + hdr_diff;
+	ctx->offset = 0;
+	ctx->processed = 0;
+}
+
 /**
  *  @brief Fragment IPv6 packet as per RFC 6282
  *
  *  @details After IPv6 compression, transmission of IPv6 over 802.15.4
  *  needs to be fragmented. Every fragment will have fragmentation header
  *  data size, data offset, data tag and payload.
  *
- *  @param Pointer to network packet
- *  @param Header difference between original IPv6 header and compressed header
+ *  @param Pointer to valid fragmentatian context
+ *  @param Pointer to varid buffer where to write the fragment
```

Comment:
Typo s/varid/valid

Reply:
and fragmentation instead fragment...meeeh...martian? :]